iceoryx_utils/include/iceoryx_utils/internal/units/duration.hpp

```diff
@@ -37,200 +38,298 @@ class Duration;
+    constexpr explicit Duration(const struct itimerspec& value) noexcept;
 
     /// @brief Construct a Duration object from std::chrono::milliseconds
-    constexpr explicit Duration(const std::chrono::milliseconds& value);
+    /// @param[in] value as milliseconds
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
+    constexpr explicit Duration(const std::chrono::milliseconds& value) noexcept;
 
     /// @brief Construct a Duration object from std::chrono::nanoseconds
-    constexpr explicit Duration(const std::chrono::nanoseconds& value);
+    /// @param[in] value as nanoseconds
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
+    constexpr explicit Duration(const std::chrono::nanoseconds& value) noexcept;
 
     /// @brief Assigns a std::chrono::milliseconds to an duration object
-    Duration& operator=(const std::chrono::milliseconds& right);
+    /// @param[in] rhs is the right hand side of the assignment
+    /// @return a reference to the Duration object with the assigned millisecond value
+    /// @attention since negative durations are not allowed, the duration will be capped to 0
+    Duration& operator=(const std::chrono::milliseconds& rhs) noexcept;
+
+    // END CONSTRUCTORS AND ASSIGNMENT
+
+    // BEGIN COMPARISON
+
+    /// @brief Equal to operator
+    /// @param[in] rhs is the right hand side of the comparison
+    /// @return true if duration equal to rhs
+    constexpr bool operator==(const Duration& rhs) const noexcept;
+
+    /// @brief Not equal to operator
+    /// @param[in] rhs is the right hand side of the comparison
+    /// @return true if duration not equal to rhs
+    constexpr bool operator!=(const Duration& rhs) const noexcept;
+
+    /// @brief Less than operator
+    /// @param[in] rhs is the right hand side of the comparison
+    /// @return true if duration is less than rhs
+    constexpr bool operator<(const Duration& rhs) const noexcept;
+
+    /// @brief Less than or equal to operator
+    /// @param[in] rhs is the right hand side of the comparison
+    /// @return true if duration is less than or equal to rhs
+    constexpr bool operator<=(const Duration& rhs) const noexcept;
+
+    /// @brief Greater than operator
+    /// @param[in] rhs is the right hand side of the comparison
+    /// @return true if duration is greater than rhs
+    constexpr bool operator>(const Duration& rhs) const noexcept;
+
+    /// @brief Greater than or equal to operator
+    /// @param[in] rhs is the right hand side of the comparison
+    /// @return true if duration is greater than or equal to rhs
+    constexpr bool operator>=(const Duration& rhs) const noexcept;
+
+    // END COMPARISON
+
+    // BEGIN ARITHMETIC
+
+    /// @brief Creates Duration object by addition
+    /// @param[in] rhs is the second summand
+    /// @return a new Duration object
+    constexpr Duration operator+(const Duration& rhs) const noexcept;
```

Comment:
Oh ... what about `operator+=`, `operator-=` and `operator*=`?
These are my most favorite operators :D

Reply:
That's a new feature, this PR just replaces the internal double representation with a fixed point one. Feel free to create a PR with your most favorite operator :P